### PR TITLE
add script to show host breakdown by OS and client version

### PR DESCRIPTION
**Pull request by davidpanderson**

### CODE REVIEW

File: html/user/host_stats.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+require_once("../inc/util.inc");
+require_once("../inc/cache.inc");
+
+$min_credit = .1;   // only count hosts with this much RAC
+$total_eac = 0;
+
+function show_type($s, $x) {
+    global $total_eac;
+    $p = number_format(100*$x->c/$total_eac, 4);
```
Comment (TheAspens) [inline html/user/host_stats.php]: On a new project (admittedly a edge case), this causes a division by zero error and the following is output in the PHP.  

> Warning: Division by zero in /root/project/html/user/host_stats.php on line 30
> 
> Warning: Division by zero in /root/project/html/user/host_stats.php on line 30
> 
> Warning: Division by zero in /root/project/html/user/host_stats.php on line 30

Something like the following will prevent this:
```
function show_type($s, $x) {
    global $total_eac;
    if ( $total_eac == 0 ) {
        $p = 0;
    } else {
        $p = number_format(100*$x->c/$total_eac, 4);
    }
    row_array(array($s, $x->n, "$p %"));
}
```

**Comment by TheAspens** [comment]
In addition to the comment above, there are a lot of one character variable names which did not provide any meaning to me when I was reading the code.  This violates the "Names should be descriptive without being verbose" rule from the [BOINC style guide](https://boinc.berkeley.edu/trac/wiki/CodingStyle#naming)

**Comment by TheAspens** [comment]
Other than those comments, the code looks good.   Thanks for the addition.  We enjoyed looking at it on Seti@Home during the projects call today.

**Comment by davidpanderson** [comment]
Good points!  I incorporated your suggestions

**Comment by TheAspens** [comment]
Thanks for the changes.  Looks good!

**Pull request merged**